include/up-cpp/datamodel/validator/Uuid.h

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-FileType: SOURCE
+// SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef UP_CPP_DATAMODEL_VALIDATOR_UUID_H
+#define UP_CPP_DATAMODEL_VALIDATOR_UUID_H
+
+#include <uprotocol/v1/uuid.pb.h>
+
+#include <chrono>
+#include <cstdint>
+#include <stdexcept>
+
+// @brief Collection of functions for validating and inspecting UUIDs.
+namespace uprotocol::datamodel::validator::uuid {
+
+/// @brief This exception indicates that a UUID object was provided that
+///        did not contain valid UUID data.
+struct InvalidUuid : public std::invalid_argument {
+	// Inherit constructors
+	using std::invalid_argument::invalid_argument;
+
+	InvalidUuid(InvalidUuid&&) noexcept;
+	InvalidUuid& operator=(InvalidUuid&&) noexcept;
+
+	InvalidUuid(const InvalidUuid&);
+	InvalidUuid& operator=(const InvalidUuid&);
+};
+
+/// @name Validity checks
+/// @{
+/// @brief Checks if the provided UUID contains valid v6 or v8 UUID data.
+/// @returns True if the UUID has valid UUID data, false otherwise.
+bool isUuid(v1::UUID);
+
+/// @brief Checks if the provided UUID contains valid v6 UUID data.
+/// @returns True if the UUID has valid UUID v6 data, false otherwise.
+bool isV6(v1::UUID);
```

Comment:
@stevenhartley -- is v6 still on the table? I could have sworn we moved to only UUIDv8?

I remember a while back we removed v6 from `up-rust`.

Reply:
Unsure. If there is agreement, I can remove it completely

Reply:
@PLeVasseur v8 only for now. We don't need to support the older versions anymore